packages/protocol/test/governance/validators.ts

```diff
@@ -2178,10 +2178,145 @@ contract('Validators', (accounts: string[]) => {
       })
     })
 
-    describe('when the sender is not an approved address', async () => {
+    describe('when the sender is not an approved address', () => {
       it('should revert', async () => {
         await assertRevert(validators.forceDeaffiliateIfValidator(validator))
       })
     })
   })
+
+  describe('#groupMembershipInEpoch', () => {
+    const validator = accounts[0]
+    const groups = accounts.slice(1, -1)
+
+    beforeEach(async () => {
+      await registry.setAddressFor(CeloContractName.DowntimeSlasher, validator)
+      await registry.setAddressFor(CeloContractName.DoubleSigningSlasher, accounts[3])
+      await registry.setAddressFor(CeloContractName.Governance, accounts[4])
+    })
+
+    describe('when the validator is added to different groups sequentially', () => {
+      const epochs = []
+      beforeEach(async () => {
+        await registerValidator(validator)
+        for (const group of groups) {
+          await registerValidatorGroup(group)
+        }
+        for (let i = 0; i < membershipHistoryLength.plus(1).toNumber(); i++) {
+          const blockNumber = await web3.eth.getBlockNumber()
+          const epochNumber = Math.floor(blockNumber / EPOCH)
+          const blocksUntilNextEpoch = (epochNumber + 1) * EPOCH - blockNumber
+          await mineBlocks(blocksUntilNextEpoch, web3)
+          epochs.push([epochNumber + 1, i !== 4 ? groups[i] : NULL_ADDRESS])
```

Comment:
Why `i !== 4`? Would be nice to include more instances in which the validator was removed from the group, maybe `i % 3 === 0`?

Reply:
Ah, I see you do this below. IMO one test like this is sufficient, what extra coverage does this test give us?